### PR TITLE
Load test k6

### PR DESCRIPTION
**Pull request by ThanatWongsamut**
- Update replica to support concurrent
- Update rate limit
- Update body validation on thread service
- Add K6 and docker

### CODE REVIEW

**Comment by ThanatWongsamut** [comment]
Load test result on Grafana
image

K6 result
image

File: api-gateway/src/utils/sercurity.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ const applySecurityMiddleware = (app: Application) => {
   app.use(helmet());
 
   // Rate limiting to prevent DDoS attacks
-  app.use(limiter);
+  // app.use(limiter);
```
Comment (sarunnut1236) [inline api-gateway/src/utils/sercurity.ts:23]: Okay for now, but in real life, we need to enable it

Comment (sarunnut1236) [inline api-gateway/src/utils/sercurity.ts:23]: Okay for now, but in reality, we need to enable it.

File: load-test/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    healthcheck:
+      test: ["CMD", "influx", "-execute", "SHOW DATABASES"]
+      interval: 10s
+      timeout: 3s
+      retries: 3
+
+  grafana:
+    image: grafana/grafana:latest
+    ports:
+      - "3002:3000"
```
Comment (sarunnut1236) [inline load-test/docker-compose.yml:31]: does anyone use port 3001??

File: thread-service/src/server.ts [hunk 3]
```diff
@@ -177,6 +177,7 @@ server.addService(threadProto.ThreadService.service, {
             updatedAt: true,
             createdAt: true,
             isDeleted: true,
+            authorId: true,
```
Comment (sarunnut1236) [inline thread-service/src/server.ts:180]: where this line came from?

**Review by sarunnut1236 — COMMENTED** [review]
Some questions

**Comment by sarunnut1236** [comment]
What are we gonna do with this NFR? do it, or drop it?
image

File: load-test/docker-compose.yml (hunk 2 above)
Comment (ThanatWongsamut) [reply load-test/docker-compose.yml:31]: I think frontend use 3001 for now

File: thread-service/src/server.ts (hunk 3 above)
Comment (ThanatWongsamut) [reply thread-service/src/server.ts:180]: Don't trust the author from client, we need to use it JWT token, so when we validate just ignore it

**Review by sarunnut1236 — APPROVED** [review]
LGTVHD+

**Pull request merged**